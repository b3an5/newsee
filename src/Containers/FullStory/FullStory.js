import React, { Component } from "react";
import { connect } from "react-redux";
import "./FullStory.scss";

class FullStory extends Component {
  constructor() {
    super();
    this.state = {
      article: {}
    };
  }

  componentDidMount = () => {
    if (
      this.props.match.params.id.includes(
        JSON.parse(localStorage.article).publishedAt
      )
    ) {
      this.setState({ article: JSON.parse(localStorage.article) });
    } else {
      this.grabArticleContent(this.props.currentStory.url);
    }
  };

  grabArticleContent = async articleUrl => {
    const {
      title,
      urlToImage,
      author,
      url,
      publishedAt
    } = this.props.currentStory;
    const fetchUrl = `https://api.diffbot.com/v3/article?token=27b09f6cb2a8e2ba60bf2717c2e9326f&url=${articleUrl}`;
    const response = await fetch(fetchUrl);
    const result = await response.json();
    if (!result.objects[0].text) {
      return alert("sorry");
    }
    const currentArticle = {
      title,
      url,
      author,
      content: result.objects[0].text,
      urlToImage,
      publishedAt
    };
    localStorage.setItem("article", JSON.stringify(currentArticle));
    this.setState({ article: currentArticle });
  };

  render() {
    const { title, urlToImage, content, author, url } = this.state.article;

    if (this.state.article.title) {
      return (
        <main className="full-article">
          <button>Favorite</button>
          <img src={urlToImage} alt={title} />
          <h1>{title}</h1>
          <h4>Author: {author}</h4>
          <p>{content}</p>
          <a href={url}>read the original article here</a>
        </main>
      );
    } else {
      return (
        <main className="full-story-loading">
          <img
            src="https://cdn.dribbble.com/users/5661/screenshots/2491233/loading-gif-800x600.gif"
            alt="loading gif"
          />
        </main>
      );
    }
  }
}

export const mapStateToProps = state => ({
  currentStory: state.currentStory
});

export default connect(mapStateToProps)(FullStory);
