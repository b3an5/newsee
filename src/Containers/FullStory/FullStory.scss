.full-article {
  display: flex;
  flex-direction: column;
  text-align: center;
  width: 80vw;
  align-items: center;
  margin: 10px auto;
  border: 3px solid black;
  padding: 15px;
  background-color: linen;
}

.full-story-loading {
  text-align: center;
}
