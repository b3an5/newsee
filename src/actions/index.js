export const saveTopStories = article => ({
  type: 'SAVE_TOP_STORIES',
  article,
})

export const saveSearchStories = article => ({
  type: 'SAVE_SEARCH_STORIES',
  article,
})

export const saveCurrentStory = article => ({
  type: 'SAVE_CURRENT_STORY',
  article,
})
