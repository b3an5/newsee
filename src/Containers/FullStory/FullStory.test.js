import { mapDispatchToProps, FullStory, mapStateToProps } from "./FullStory";
import { saveTopStories, saveSearchStories } from "../../actions/index";
import { shallow } from "enzyme";
import {
  mockCurrentStory,
  mockTopStories,
  mockSearchStories
} from "../../utilities/mockData";
import React from "react";

describe("fullStory", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<FullStory currentStory={mockCurrentStory} />);
  });
  describe("fullStory component", () => {
    it("should match snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should match default state", () => {
      expect(wrapper.state()).toEqual({
        article: {}
      });
    });

    it("should call favorite article when button is pressed", () => {
      wrapper.state({ article: mockCurrentStory });
      expect(wrapper.state()).toEqual({
        article: {}
      });
      // wrapper.find(".fav-butt").simulate("click");
    });
  });

  describe("mstp", () => {
    it("should return an object with topstories and searchstories", () => {
      const mockState = {
        topStories: mockTopStories,
        searchStories: mockSearchStories,
        currentStory: mockCurrentStory
      };
      const expected = {
        currentStory: mockCurrentStory
      };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });
});
