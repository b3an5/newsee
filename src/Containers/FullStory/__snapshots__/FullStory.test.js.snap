// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fullStory fullStory component should match loading snapshot 1`] = `ShallowWrapper {}`;

exports[`fullStory fullStory component should match snapshot 1`] = `ShallowWrapper {}`;
